pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -120,9 +121,52 @@ func (bw *quotaRestrictedBlobWriter) Commit(ctx context.Context, provisional dis
 	return can, err
 }
 
-// admitBlobWrite checks whether the blob does not exceed image quota, if set. Returns
+// admitBlobWrite checks whether the blob does not exceed image quota or limit ranges, if set. Returns
 // ErrAccessDenied error if the quota is exceeded.
-func admitBlobWrite(ctx context.Context, repo *repository) error {
+func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
+	if err := admitLimitRanges(ctx, repo, size); err != nil {
+		return err
+	}
+
+	if err := admitQuotas(ctx, repo); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// admitLimitRanges checks the blob against any established limit ranges.
+func admitLimitRanges(ctx context.Context, repo *repository, size int64) error {
+	if size < 1 {
+		return nil
+	}
+
+	limitranges, err := repo.limitClient.LimitRanges(repo.namespace).List(kapi.ListOptions{})
+	if err != nil {
+		if kerrors.IsForbidden(err) {
```

Comment:
Well, if you have your registry pinned to specific version, and if you're doing prod, that's the only reasonable solution, then only upon updating the image, iow. manually updating docker-registry DC, you can opt-out of the quota enforcement. Imho, that's would be reasonable approach, the only thing is to document it appropriately.